src/extension.ts

```diff
@@ -270,6 +271,36 @@ export async function activate(context: VSCode.ExtensionContext) {
   IssueService.init(languageClient, secondaryLocationsTree, issueLocationsView);
 
   context.subscriptions.push(languageClient.onNotification(ShowIssueNotification.type, IssueService.showIssue));
+  
+  context.subscriptions.push(languageClient.onNotification(protocol.StartProgressNotification.type, (params: protocol.StartProgressNotificationParams) => {
+    if (currentProgress) {
+      // If there's an existing progress, resolve it first
+      currentProgress.resolve();
```

Comment:
Continuing on my comment above: automatically `resolve()`-ing the current progress may not be what is expected: if the end notification was not received for a given `taskId`, the task is probably still going on.